client/overview/task-list/tasks.js

```diff
@@ -20,15 +23,57 @@ export const getTasks = ( {
 	isAccountOverviewTasksEnabled,
 	numDisputesNeedingResponse = 0,
 } ) => {
-	const { status, currentDeadline, pastDue, accountLink } = accountStatus;
+	const renderModal = (
+		errorMessages,
+		status,
+		accountLink,
+		currentDeadline
+	) => {
+		const container = document.createElement( 'div' );
+		container.id = 'wcpay-update-business-details-container';
+		document.body.appendChild( container );
+		ReactDOM.render(
+			<UpdateBusinessDetailsModal
```

Comment:
We're doing this in a really hacky way 😅 As we don't have an identifiable element, and an event to trigger the component mount. I think your approach of rendering it on task click event is okay. Although we could not show the modal on `More details` click, like in the design.

The current implementation will keep adding new containers every time we close the modal and click on the task. To fix that we need to add it only once, and provide a `key` to the component so the state get discard, and it does not remain closed. Something like:
```js
		let container = document.querySelector(
			'#wcpay-update-business-details-container'
		);
		if ( ! container ) {
			container = document.createElement( 'div' );
			container.id = 'wcpay-update-business-details-container';
			document.body.appendChild( container );
		}
		ReactDOM.render(
			<UpdateBusinessDetailsModal
				key={ Date.now() }
```

Reply:
Yeah - it's definitely not ideal 😅 thanks for the tip on not injecting a new container each time.